components/l10n/browser/default_locale_util_linux.cc

```diff
@@ -7,22 +7,30 @@
 
 #include <cstdlib>
 #include <optional>
+#include <string>
+
+#include "base/environment.h"
 
 namespace brave_l10n {
 
-std::optional<std::string> MaybeGetDefaultLocaleString() {
-  // LC_ALL should always override the LANG variable, whether it is set or not.
-  char const* language = language = std::getenv("LC_ALL");
+namespace {
 
-  if (!language || !*language) {
-    language = std::getenv("LANG");
-  }
+constexpr char kEnvVarLcAll[] = "LC_ALL";
+constexpr char kEnvVarLang[] = "LANG";
+
+}  // namespace
+
+std::optional<std::string> MaybeGetDefaultLocaleString() {
+  std::unique_ptr<base::Environment> env = base::Environment::Create();
 
-  if (!language || !*language) {
-    return std::nullopt;
+  std::string language;
+  if (!env->GetVar(kEnvVarLcAll, &language) || language.empty()) {
+    if (!env->GetVar(kEnvVarLang, &language) || language.empty()) {
+      return std::nullopt;
+    }
```

Comment:
maybe you can avoid using the old outparam interface with:

```cxx
std::optional<std::string> language;
if (language = env->GetVar(kEnvVarLcAll, &language); !language || language.empty()) {
  if (language = env->GetVar(kEnvVarLang, &language); !language || language.empty()) {
    return std::nullopt;
  }
}

Reply:
Good catch, thanks

Reply:
Fixed